metadata-service/restli-client/src/main/java/com/linkedin/entity/client/JavaEntityClient.java

```diff
@@ -285,4 +304,69 @@ public SearchResult filter(@Nonnull String entity, @Nonnull Filter filter, @Null
         int start, int count, @Nonnull String actor) throws RemoteInvocationException {
         return _entitySearchService.filter(entity, filter, sortCriterion, start, count);
     }
+
+    @SneakyThrows
+    @Override
+    public VersionedAspect getAspect(@Nonnull String urn, @Nonnull String aspect, @Nonnull Long version,
+        @Nonnull String actor) throws RemoteInvocationException {
+        return _entityService.getVersionedAspect(Urn.createFromString(urn), aspect, version);
+    }
+
+    @SneakyThrows
+    @Override
+    public VersionedAspect getAspectOrNull(@Nonnull String urn, @Nonnull String aspect, @Nonnull Long version,
+        @Nonnull String actor) throws RemoteInvocationException {
+        return _entityService.getVersionedAspect(Urn.createFromString(urn), aspect, version);
+    }
+
+    @SneakyThrows
+    @Override
+    public List<EnvelopedAspect> getTimeseriesAspectValues(@Nonnull String urn, @Nonnull String entity,
+        @Nonnull String aspect, @Nullable Long startTimeMillis, @Nullable Long endTimeMillis, @Nullable Integer limit,
+        @Nullable String actor) throws RemoteInvocationException {
+        GetTimeseriesAspectValuesResponse response = new GetTimeseriesAspectValuesResponse();
+        response.setEntityName(entity);
+        response.setAspectName(aspect);
+        if (startTimeMillis != null) {
+            response.setStartTimeMillis(startTimeMillis);
+        }
+        if (endTimeMillis != null) {
+            response.setEndTimeMillis(endTimeMillis);
+        }
+        response.setLimit(limit);
+        response.setValues(new EnvelopedAspectArray(
+            _timeseriesAspectService.getAspectValues(Urn.createFromString(urn), entity, aspect, startTimeMillis, endTimeMillis,
+                limit)));
+        return response.getValues();
+    }
+
+    // TODO: Factor out ingest logic into a util that can be accessed by the java client and the resource
+    @SneakyThrows
+    @Override
+    public String ingestProposal(@Nonnull MetadataChangeProposal metadataChangeProposal,
+        @Nonnull String actor) throws RemoteInvocationException {
+        final AuditStamp auditStamp =
+            new AuditStamp().setTime(_clock.millis()).setActor(Urn.createFromString(Constants.UNKNOWN_ACTOR));
+        final List<MetadataChangeProposal> additionalChanges =
+            AspectUtils.getAdditionalChanges(metadataChangeProposal, _entityService);
+
+        Urn urn = _entityService.ingestProposal(metadataChangeProposal, auditStamp);
+        additionalChanges.forEach(proposal -> _entityService.ingestProposal(proposal, auditStamp));
+        return urn.toString();
+    }
+
+    @SneakyThrows
+    @Override
+    public <T extends RecordTemplate> Optional<T> getVersionedAspect(@Nonnull String urn, @Nonnull String aspect,
```

Comment:
Just curious where is this used?

Reply:
Looks like just here: https://github.com/linkedin/datahub/blob/d7a6ca542351d9b860378917f31b0dff1bc3d9e5/datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/SubTypesResolver.java

I think as a followup we can converge some of these get aspect methods... they are a bit redundant